ramp4-pcar4/ramp4-pcar4
# Docsite updates

### Related Item(s)
- #1751
- #1766
- #1750

### Changes
- Updated README to provide link to docsite (#1766) and added commands to preview/build the docsite locally
- Vitepress Docs:
  - Replaced placeholder cards in hero page with quick links to docs sections
  - Replaced nav bar logo with non-text version
  - Move the version number to the page title and provided direct link to releases page instead of using a dropdown
  - Sidebar now provides some flow through the docs
  - Updated usages of "R4MP" to "RAMP4" to support screen-readers
- TS Docs:
  - Uses the same background color and link color as the Vitepress docs
  - Create a separate README file for the ts docs homepage
  - Hid the items mentioned in https://github.com/ramp4-pcar4/ramp4-pcar4/issues/1750

### Notes
- Pending work:
  - #1752: update code to hide certain properties from docsite
  - From Slack: Make sure classes have a JS Doc especially if it is inheriting a class, otherwise TypeDoc will just use the super-class JS Doc which can be confusing to the reader
  - Need to update markdown docs since most of it is out of date
    - Update dead links in markdown docs (remove `ignoreDeadLinks` in `.vitepress/config.ts` and build the vite docs to see all the dead links)
  - Update code references in markdown docs to link to ts api docs
  - Update uses of `any` type if possible

### Testing

The docsite can be built locally with the following steps:

- Vitepress:
  - Run `npx vitepress dev docs` and open the link provided in the console 
- TypeDoc:
  - Run `npm run build:ts-docs` and open `index.html` in the generated `ts-docs` folder

Note that some links will seem broken since they are hard-coded to link to the docsite built by the main repo.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/1806)
